### PR TITLE
[5.1] Adds automatic Chrome version detection

### PR DESCRIPTION
**Pull request by peterfox**
Adds the `—detect` option for `dusk:chrome-driver` that can be provided a custom path or it will use an expected default for the operating system. It will then try to automatically detect the version of Chrome being used and then download the correct chrome driver for that version. Will make it easier for developers to side step having to check which version of Chrome they have as the current default is for the command to simply grab the latest driver version but sometimes this will not line up with what they have installed.

Haven't tested it for Windows or Linux due to a lack of access to those OSs to test on, I can't see there being a problem with Linux but it's hard to tell with Windows and I just don't have a machine to hand to test it with.

### CODE REVIEW

**Comment by staudenmeir** [comment]
Unfortunately, this is more complicated.

`--version` doesn't work on Windows: https://bugs.chromium.org/p/chromium/issues/detail?id=158372

We can use `reg query "HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon" /v version` instead:

```
HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon
    version    REG_SZ    74.0.3729.131
```

You can use Homestead to test it on Linux.

We should also support Google Chrome: `/usr/bin/google-chrome`

File: src/Console/ChromeDriverCommand.php [hunk 1]
```diff
@@ -127,15 +141,44 @@ public function handle()
         $this->info(sprintf($message, $all ? 'binaries' : 'binary', $version));
     }
 
+    protected function detectVersion($currentOS, $path = null)
```
Comment (staudenmeir) [inline src/Console/ChromeDriverCommand.php]: Please add the PHPDoc.

File: src/Console/ChromeDriverCommand.php [hunk 2]
```diff
@@ -127,15 +141,44 @@ public function handle()
...
+    {
+        $installPath = $path ? realpath($path) : $this->installPaths[$currentOS];
+
+        if (! file_exists($installPath) || is_dir($installPath) || ! is_executable($installPath)) {
```
Comment (staudenmeir) [inline src/Console/ChromeDriverCommand.php]: Do we need the `is_dir()` check? Doesn't `file_exists()` cover that?

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by staudenmeir** [comment]
I've added the Chrome version detection to my updater package: [staudenmeir/dusk-updater](https://github.com/staudenmeir/dusk-updater)

**Comment by browner12** [comment]
Can we re-address this and start by making it work for Unix systems?  Seems like we have that part solved.  It seems silly to hold this back because Windows doesn't work.

**Pull request closed without merging**